plugins/action_send_notification/src/sendNotification.ts

```diff
@@ -18,10 +32,20 @@ const sendNotification: ActionPluginType = async ({ parameters, applicationData,
   const db = databaseMethods(DBConnect)
 
   const {
-    environmentData: { appRootFolder, filesFolder, SMTPConfig },
+    environmentData: { appRootFolder, filesFolder, SMTPConfig, webHostUrl, productionHost },
     other,
   } = applicationData as ActionApplicationData
 
+  const testingEmail = config.testingEmail
+
+  const mode = getOperationMode({
+    mailHog: USE_MAIL_HOG,
+    webHostUrl,
+    productionHost,
+    suppressEmail: other?.suppressEmail ?? false,
+    testingEmail,
+  })
```

Comment:
The logic for each kind of state was getting a bit messy, so I've collapsed them all down to one of four "modes", which makes it easier to follow what situation we're in for each conditional branch.